### ISSUE TITLE
Building deps for Cairo fails

### ISSUE THREAD

**Issue by ZacCranko** [staticfloat/homebrew-juliadeps#63]
Hi there,

Getting this when I try to build Cairo...

```julia
julia> Pkg.build("Cairo")
INFO: Building Homebrew
INFO: Recompiling stale cache file /Users/zaccranko/.julia/lib/v0.4/URIParser.ji for module URIParser.
INFO: Recompiling stale cache file /Users/zaccranko/.julia/lib/v0.4/SHA.ji for module SHA.
remote: Counting objects: 175, done.
remote: Compressing objects: 100% (95/95), done.
remote: Total 175 (delta 138), reused 117 (delta 80), pack-reused 0
Receiving objects: 100% (175/175), 20.90 KiB | 0 bytes/s, done.
Resolving deltas: 100% (138/138), completed with 20 local objects.
From https://github.com/Homebrew/homebrew
   4fbd40a..8b71e0f  master     -> origin/master
HEAD is now at 8b71e0f rocksdb: update 3.13 bottle.
HEAD is now at c490b44 Merge pull request #61 from staticfloat/staging
INFO: Building Cairo
Warning: You are using OS X 10.11.
We do not provide support for this pre-release version.
You may encounter build failures or other breakage.
==> Installing cairo from staticfloat/homebrew-juliadeps
==> Installing dependencies for staticfloat/juliadeps/cairo: staticfloat/j
==> Installing staticfloat/juliadeps/cairo dependency: staticfloat/juliade
==> Downloading http://cairographics.org/releases/pixman-0.32.6.tar.gz
Already downloaded: /Users/zaccranko/Library/Caches/Homebrew.jl/pixman-0.32.6.tar.gz
==> ./configure --disable-silent-rules --prefix=/Users/zaccranko/.julia/v0.4/Home
==> make install
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -g -O2 -Wall -Wdeclaration-after-statement -fno-strict-aliasing -fvisibility=hidden -c pixman-region16.c -o pixman-region16.o >/dev/null 2>&1
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -g -O2 -Wall -Wdeclaration-after-statement -fno-strict-aliasing -fvisibility=hidden -c pixman-region32.c -o pixman-region32.o >/dev/null 2>&1
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -g -O2 -Wall -Wdeclaration-after-statement -fno-strict-aliasing -fvisibility=hidden -c pixman-fast-path.c -o pixman-fast-path.o >/dev/null 2>&1
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -msse2 -Winline -g -O2 -Wall -Wdeclaration-after-statement -fno-strict-aliasing -fvisibility=hidden -c pixman-sse2.c -o libpixman_sse2_la-pixman-sse2.o >/dev/null 2>&1
make: *** [install-recursive] Error 1

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/homebrew):
  https://github.com/staticfloat/homebrew-juliadeps/issues

Warning: You are using OS X 10.11.
We do not provide support for this pre-release version.
You may encounter build failures or other breakage.
================================[ ERROR: Cairo ]=================================

LoadError: failed process: Process(`/Users/zaccranko/.julia/v0.4/Homebrew/deps/usr/bin/brew install --force-bottle staticfloat/juliadeps/cairo`, ProcessExited(1)) [1]
while loading /Users/zaccranko/.julia/v0.4/Cairo/deps/build.jl, in expression starting on line 144

=================================================================================

================================[ BUILD ERRORS ]=================================

WARNING: Cairo had build errors.

 - packages with build errors remain installed in /Users/zaccranko/.julia/v0.4
 - build the package(s) and all dependencies with `Pkg.build("Cairo")`
 - build a single package by running its `deps/build.jl` script

=================================================================================
```

**Comment by staticfloat** [comment]
This is also because of the following warning:

```
Warning: You are using OS X 10.11.
We do not provide support for this pre-release version.
You may encounter build failures or other breakage.
```

In short, because OSX 10.11 is still in prerelease, we are not building bottles for it yet.  This means that there are no prebuilt binaries downloadable, and everything must be installed from source on your machine, which isn't really the use case we're shooting for with a lot of these formulae in `homebrew-juliadeps`.  It's hard to see what went wrong here, but I would suggest running `brew install cairo pango` outside of Julia to install `cairo` and `pango` system-wide, and seeing if that allows `Pkg.build("Cairo")` to work.

**Comment by ZacCranko** [comment]
Ah fair enough, this explains the other issue I was having, which I'll remove then. Sorry about that.

**Comment by bosmacs** [comment]
As of a couple days ago, bottles are available for cairo and pango on El Capitan.

**Comment by staticfloat** [comment]
Yes, unfortunately they are not `cellar: any`.  I will start building bottles for El Capitan once it's actually out and I create a new VM for it.

**Comment by mseri** [comment]
I submitted a PR that should fix the problem with pixman and cairo.

**Comment by lewisl** [comment]
@mseri Not clear it's fixed.  I just tried to install for julia 0.4.0-rc4 and the cairo build failed on pixman:

==> Installing staticfloat/juliadeps/cairo dependency: staticfloat/juliad
==> Downloading http://cairographics.org/releases/pixman-0.32.6.tar.gz
Already downloaded: /Users/lewislevinmbr/Library/Caches/Homebrew.jl/pixman-0.32.6.tar.gz
==> ./configure --disable-silent-rules --prefix=/Users/lewislevinmbr/.julia/v0.4
==> make install
Last 15 lines from /Users/lewislevinmbr/Library/Logs/Homebrew/pixman/02.make:
/bin/sh ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..     -g -O2 -Wall -Wdeclaration-after-statement -fno-strict-aliasing -fvisibility=hidden -c -o pixman-utils.lo pixman-utils.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -g -O2 -Wall -Wdeclaration-after-statement -fno-strict-aliasing -fvisibility=hidden -c pixman-utils.c  -fno-common -DPIC -o .libs/pixman-utils.o
/bin/sh ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..    -mmmx -Winline -g -O2 -Wall -Wdeclaration-after-statement -fno-strict-aliasing -fvisibility=hidden -c -o libpixman_mmx_la-pixman-mmx.lo `test -f 'pixman-mmx.c' || echo './'`pixman-mmx.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -mmmx -Winline -g -O2 -Wall -Wdeclaration-after-statement -fno-strict-aliasing -fvisibility=hidden -c pixman-mmx.c  -fno-common -DPIC -o .libs/libpixman_mmx_la-pixman-mmx.o
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -g -O2 -Wall -Wdeclaration-after-statement -fno-strict-aliasing -fvisibility=hidden -c pixman-utils.c -o pixman-utils.o >/dev/null 2>&1
/bin/sh ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..    -msse2 -Winline -g -O2 -Wall -Wdeclaration-after-statement -fno-strict-aliasing -fvisibility=hidden -c -o libpixman_sse2_la-pixman-sse2.lo `test -f 'pixman-sse2.c' || echo './'`pixman-sse2.c
pixman-mmx.c:100:20: error: constraint 'K' expects an integer constant expression
        : "y" (__A), "K" (__N)
                          ^~~
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -msse2 -Winline -g -O2 -Wall -Wdeclaration-after-statement -fno-strict-aliasing -fvisibility=hidden -c pixman-sse2.c  -fno-common -DPIC -o .libs/libpixman_sse2_la-pixman-sse2.o
1 error generated.
make[1]: *** [libpixman_mmx_la-pixman-mmx.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -msse2 -Winline -g -O2 -Wall -Wdeclaration-after-statement -fno-strict-aliasing -fvisibility=hidden -c pixman-sse2.c -o libpixman_sse2_la-pixman-sse2.o >/dev/null 2>&1
make: *** [install-recursive] Error 1

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/homebrew):
  https://github.com/staticfloat/homebrew-juliadeps/issues

Should I just wait a bit?  I see it's a private release; not actually released.

**Comment by staticfloat** [comment]
The fix has been merged into staging but not master yet. Give it a day or
two, I'm planning on building bottles for el Capitan soon.

**Comment by neronmoon** [comment]
+1

**Comment by staticfloat** [comment]
Hey guys, sorry for the wait.  I have an El Capitan buildbot up, and I'm still testing to make sure everything is on the up-and-up.  Could those of you with access to a 10.11 machine that are willing to help me test try to install the latest `julia` from [my tap](https://github.com/staticfloat/homebrew-julia)?  This should be as simple as `brew tap staticfloat/julia; brew install staticfloat/julia/julia`.

**Comment by goedman** [comment]
Did above brew steps without any problems and Ipopt now builds and installs fine. Thanks a lot!

**Comment by staticfloat** [comment]
I have now merged a big upgrade for OSX 10.11.  Please `using Homebrew; Homebrew.update()` and try again.

**Comment by goedman** [comment]
Again, works flawlessly for me. Thanks a lot Elliot!
